com.unity.netcode.gameobjects/Components/NetworkTransform.cs

```diff
@@ -523,76 +546,130 @@ internal bool ApplyTransformToNetworkState(ref NetworkTransformState networkStat
             //  this still is overly costly and could use more improvements.
             //
             // (ditto for scale components)
-            if (SyncPositionX &&
-                Mathf.Abs(networkState.PositionX - position.x) > PositionThreshold)
+            if (SyncPositionX)
             {
-                networkState.PositionX = position.x;
-                networkState.HasPositionX = true;
-                isPositionDirty = true;
+                if (Mathf.Abs(networkState.PositionX - position.x) > PositionThreshold)
+                {
+                    networkState.PositionX = position.x;
+                    networkState.HasPositionX = true;
+                    isPositionDirty = true;
+                }
+                else
+                {
+                    networkState.HasPositionX = false;
```

Comment:
Is it safer (and cleaner to read) to set this unconditionally and then change it only if there is a substantial change?
```
networkState.HasPositionX = false;
if (syncposx && abovethreshold)
{
  haspositionx = true
}
```

Would safeguard against someone changing the SyncPositionX from true to false and having HasPositionX be stuck true with an old value? Not sure if that's an actual scenario that can happen - I'm not familiar enough with the code yet.

Reply:
From the non-authority side it means that by disabling SyncPositionX all updates to the x axis would cease for that instance.  Upon re-enabling SyncPositionX, the HasPositionX value would be re-synchronized (the new state would either have a bitset or not) and will be reset at that time.

For this PR I am going to say that we might just keep it the way it is.  PR #2110 has some more clean up and additional fixes that help with the readability.